File: java/src/com/alofmethbin/jadawin/styles/TextIndex.java
package com.alofmethbin.jadawin.styles;

import com.alofmethbin.jadawin.Article;

public class TextIndex extends BaseStyle {

    @Override
    public String indexStyle(Article article) {
        return article.hasContent() ? "document" : super.indexStyle(article);
    }
}

File: ruby/styles/text_index.rb
require_relative '../link'

module Styles
  class TextIndex < Events
    def render( compiler, parents, article, data)
      data['index_style']     = 'document' if article.has_any_content?
      data['text_index']      = article.text_index( parents[-1].children)
    end
  end
end
